packages/did/src/Did.utils.ts

```diff
@@ -403,6 +414,28 @@ export async function writeDidFromPublicKeys(
   return { extrinsic, did: getKiltDidFromIdentifier(didIdentifier, 'full') }
 }
 
+export async function writeDidFromPublicKeysAndServices(
+  signer: KeystoreSigner,
+  submitter: IIdentity['address'],
+  publicKeys: PublicKeyRoleAssignment,
+  endpoints: IDidServiceEndpoint[]
+): Promise<{ extrinsic: SubmittableExtrinsic; did: string }> {
+  const { [KeyRelationship.authentication]: authenticationKey } = publicKeys
+  if (!authenticationKey)
+    throw Error(`${KeyRelationship.authentication} key is required`)
+  const didIdentifier = encodeAddress(authenticationKey.publicKey, 38)
+  const extrinsic = await generateCreateTx({
+    signer,
+    submitter,
+    didIdentifier,
+    keys: publicKeys,
+    alg: getSignatureAlgForKeyType(authenticationKey.type),
+    signingPublicKey: authenticationKey.publicKey,
+    endpoints,
+  })
+  return { extrinsic, did: getKiltDidFromIdentifier(didIdentifier, 'full') }
```

Comment:
Here you return the function `getKiltDidFromIdentifier` but other places you make a variable and then you return it.  I would recommend handling this in the same way.

Reply:
Good catch! https://github.com/KILTprotocol/sdk-js/pull/428/commits/bfc0773f65e9742d7ca2ce1333800857ffe8d273